components/builder-graph/src/package_graph_target.rs

```diff
@@ -464,83 +471,131 @@ impl PackageGraphForTarget {
+        // compute/filter the graph. For example if we use touched to indicate things in core that
+        // have been promoted, we want to propagate the updates fully before filtering the
+        // graph to a single origin
+        let mut preconditioned_graph = self.precondition_graph();
+
+        // Some things in the touched set might be missing from the graph, but still buildable
+        // (e.g first time build of a new plan) So we insert them here. Unbuildable will help us
+        // determine which things lack a plan linkage and should be dropped.
+        for &package in touched.iter() {
+            preconditioned_graph.add_node(package);
+        }
 
-        let touched: Vec<PackageIdentIntern> = touched.iter().map(|x| x.into()).collect();
-        let rebuild_set = graph_helpers::compute_rebuild_set(&preconditioned_graph,
-                                                             unbuildable,
-                                                             &touched,
-                                                             origin,
-                                                             self.target);
+        let (rebuild_set, unbuildable_reasons) =
+            graph_helpers::compute_rebuild_set(&preconditioned_graph,
+                                               unbuildable,
+                                               &touched,
+                                               origin,
+                                               self.target);
 
         // TODO DO check of rebuild set to make sure that it includes the pinned versions that had
         // edges added in the precondition_graph phase above.
 
         debug!("Rebuild: {} {}\n",
                rebuild_set.len(),
                join_idents(", ", &rebuild_set));
+        println!("PRECOND GRAPH:\n{:?}", preconditioned_graph);
 
         let build_order = graph_helpers::compute_build_order(&preconditioned_graph, &rebuild_set);
         // Rework this later
         debug!("CB: {} components", build_order.len());
         for component in &build_order {
             debug!("CB: #{} {}", component.len(), join_idents(", ", component));
         }
+        println!("BUILD ORDER:\n{:?}", build_order);
 
-        let mut latest = HashMap::<PackageIdent, PackageIdent>::new();
-        for ident in base_set {
-            latest.insert(short_ident(&ident, false), ident.clone());
-        }
+        let mut latest = HashMap::<PackageIdentIntern, UnresolvedPackageIdent>::new();
 
-        let mut file = File::create("latest_from_base.txt").expect("Failed to initialize file");
-        for (k, v) in &latest {
-            file.write_all(format!("{}: {}\n", &k, &v).as_bytes())
-                .unwrap();
-        }
-
-        let mut built: Vec<PackageBuild> = Vec::new();
+        let mut build_graph: DiGraphMap<UnresolvedPackageIdent, EdgeType> = DiGraphMap::new();
+        //        let mut built: Vec<PackageBuild> = Vec::new();
```

Comment:
We can probably remove this commented out variable as it appears to be a vestige of a previous development iteration.